Missing include for function

Compiling with clang++
```
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
gives the error 
```
expected-lite/include/nonstd/expected.hpp:1207:8: error: explicit specialization of non-template struct 'hash'
struct hash< nonstd::expected<T,E> >
       ^   ~~~~~~~~~~~~~~~~~~~~~~~~~
```
looks like `expected.hpp` is missing a `#include <functional>` which is where `std::hash` is [defined](http://en.cppreference.com/w/cpp/utility/hash).

--- a/include/nonstd/expected.hpp
+++ b/include/nonstd/expected.hpp
@@ -13,6 +13,7 @@
 
 #include <cassert>
 #include <exception>
+#include <functional>
 #include <initializer_list>
 #include <new>
 #include <stdexcept>

